### PR TITLE
Skip all integration tests if disabled

### PR DESCRIPTION
**Pull request by melvo**
PR check list:
- [x] Document your code
- [x] Add `\since QXmpp 1.X`
- [x] Fix doxygen warnings (see log when building with `-DBUILD_DOCUMENTATION=ON`)
- [x] Update `doc/xep.doc`
- [x] Add unit tests
- [x] Format the code: Run `clang-format -i src/<edited-file(s)> tests/<edited-file(s)>`

### CODE REVIEW

**Comment by lnjX** [comment]
That's no integration test. I know that these tests can cause problems, but marking them as "integration tests" is questionable.

File: tests/qxmppiceconnection/tst_qxmppiceconnection.cpp [hunk 1]
```diff
@@ -19,6 +20,8 @@ private slots:
 
 void tst_QXmppIceConnection::testBind()
 {
+    SKIP_IF_INTEGRATION_TESTS_DISABLED();
```
Comment (lnjX) [inline tests/qxmppiceconnection/tst_qxmppiceconnection.cpp:23]: just a note: `SKIP_IF_INTEGRATION_TESTS_DISABLED()` doesn't need an extra `;` at the end

**Comment by melvo** [comment]
> That's no integration test. I know that these tests can cause problems, but marking them as "integration tests" is questionable.

Thanks. How could that problem be solved?

**Comment by lnjX** [comment]
> Thanks. How could that problem be solved?

Fixing the tests, marking tests requiring a working internet connection as online tests, make online tests optional.

**Pull request closed without merging**